src/ToolInfo.php

```diff
@@ -26,67 +26,92 @@
     const COMPOSER_INSTALLED_FILE = '/../../composer/installed.json';
     const COMPOSER_PACKAGE_NAME = 'friendsofphp/php-cs-fixer';
 
-    public static function getComposerVersion()
+    /**
+     * @var string
+     */
+    private $script;
+
+    /**
+     * @var null|string
+     */
+    private $composerVersion;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledAsPhar;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledByComposer;
+
+    /**
+     * @var null|string
+     */
+    private $scriptDir;
+
+    /**
+     * @param null|string $script
+     */
+    public function __construct($script)
     {
-        static $result;
+        $this->script = (null !== $script ? $script : $_SERVER['SCRIPT_NAME']);
```

Comment:
It's pretty obvious to me that ternary operator will be evaluated before the assignment as the opposite would almost defeat the purpose of the ternary operator, but I agree parentheses still help for readability. What do others think?